lib/stripe/subscriptions/usage.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Stripe.SubscriptionItem.Usage do
+  @moduledoc """
+  Work with Stripe usage record objects.
+
+  Stripe API reference: https://stripe.com/docs/api/usage_records
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+
+  @type t :: %__MODULE__{
+          id: Stripe.id(),
+          object: String.t(),
+          invoice: Stripe.id() | nil,
+          livemode: boolean,
+          quantity: non_neg_integer,
+          subscription_item: Stripe.id() | Stripe.SubscriptionItem.t(),
+          timestamp: Stripe.timestamp()
+        }
+
+  defstruct [
+    :id,
+    :object,
+    :invoice,
+    :livemode,
+    :quantity,
+    :subscription_item,
+    :timestamp
+  ]
+
+  @plural_endpoint "subscription_items"
+
+  @doc """
+  Creates a usage record for a specified subscription item id and date, and fills it with a quantity.
+  """
+  def create(id, params, opts \\ [])
+
+  @doc """
+  Creates a usage record for a specified subscription item id and date, and fills it with a quantity.
+  """
+  @spec create(Stripe.id(), params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params: %{
+              :quantity => float,
+              :timestamp => Stripe.timestamp() | non_neg_integer,
+              optional(:action) => String.t()
+             } 
+  def create(id, params, opts) do
```

Comment:
We should probably default `opts` to `opts \\ []` in case nobody passes the 3rd arg.

Reply:
I think the case where the `opts` is not passed in is handled by the header on line 36.

Reply:
oh yep you are right!